app/controllers/agent_connect_controller.rb

```diff
@@ -115,38 +115,76 @@ def connect_user(callback_client)
     redirect_to after_sign_in_path_for(user)
   end
 
+  # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
   def connect_agent(callback_client)
-    # Agent Connect recommande de faire la réconciliation sur l'email et non pas sur le sub
-    # voir https://github.com/numerique-gouv/agentconnect-documentation/blob/main/doc_fs/donnees_fournies.md#le-champ-sub
-    agent = Agent.active.find_by(email: callback_client.user_email)
+    sub = callback_client.openid_sub
+    agent_by_sub = Agent.active.find_by(pro_connect_openid_sub: sub)
+    agent_by_email = Agent.active.find_by(email: callback_client.user_email)
+    Sentry.set_user({ email: callback_client.user_email })
+
+    # On trouve un agent vie le sub, et un autre agent via l'e-mail
+    if agent_by_email && agent_by_sub && agent_by_email != agent_by_sub
+      error_message = <<~ERROR
+        Il existe deux comptes correspondant : #{agent_by_email.email} et #{agent_by_sub.email}.
+        Nous vous invitons à contacter le support à l'adresse <a href='mailto:#{current_domain.support_email}'>#{current_domain.support_email}</a>.
+      ERROR
+      Sentry.capture_message(error_message, extra: { user_info: callback_client.user_info })
+      flash[:error] = error_message
+      redirect_to new_agent_session_path and return
+    end
 
-    if current_domain.allow_self_onboarding
-      agent ||= Agent.new(email: callback_client.user_email, password: SecureRandom.base64(32))
+    # On trouve un agent par e-mail et cet agent porte un autre sub.
+    if agent_by_email&.pro_connect_openid_sub && agent_by_email.pro_connect_openid_sub != sub
+      error_message = <<~ERROR
+        Ce compte ProConnect est lié à l'adresse e-mail #{callback_client.user_email}.<br />
+        Or cette adresse est déjà liée à compte existant qui est connecté avec un autre compte ProConnect.<br />
```

Comment:
Je ne sais pas si c’est mieux, mais une suggestion : 

(en tout cas il manque un « un » dans ta tournure )

```suggestion
        Un compte agent existe déjà sur #{current_domain.name} pour cette adresse email, cependant il est lié à un autre compte ProConnect.<br />
```